packages/testcontainers/src/containers/DeFiDContainer.ts

```diff
@@ -246,7 +235,9 @@ export abstract class DeFiDContainer extends DockerContainer {
    * @param {number} [timeout=30000] in millis
    */
   async restart (timeout: number = 30000): Promise<void> {
-    await this.container?.restart()
+    const dockerrode = (await dockerClient).dockerode
+    const dockerrodeContainer = dockerrode.getContainer(this.requireContainer().getId())
+    await dockerrodeContainer.restart()
```

Comment:
seems you're trying use `dockerode` attach the running docker to execute `restart` since testcontainer doesn't support `restart` <-- gosh!!

Reply:
and seems we get advised to use `stop and start` instead of `restart` in testcontainer
https://github.com/testcontainers/testcontainers-node/issues/83#issuecomment-638379724

Reply:
Yes I saw this, but for this case, where its setting the `defi.conf` file for masternode, the restart is required to retain state inside the container, and stopping and then starting it would cause it to lose all state. I also tried working with stopping and starting with dockerrode instead of testcontainer, but for some reason it doesn't work.

Reply:
Yes I saw this, but for this case, where its setting the `defi.conf` file for masternode, the restart is required to retain state inside the container, and stopping and then starting it would cause it to lose all state. I also tried working with stopping and starting with dockerrode instead of testcontainer, but for some reason it doesn't work.

Reply:
hmm... perhaps we need docker volume to retain the state??

Reply:
default api provided by `testcontainer-node` does not support volumnes, but does support bind mount and tmpfs. I think if we need access to volumes we'll have to use the docker-compose apis provided by them, which needs a docker-compose file to be provided.

Reply:
hmm... do you think `testcontainer` suit our case?? i feel its a little bit `workaround`.. :thinking: 
cc @fuxingloh